Correct description of controller/built-in node (#5812)
jenkins-infra/jenkins.io: content/doc/book/managing/nodes.adoc

A node is taken offline if any of these values
go outside the configured threshold.

> Jenkins supports two types of nodes: _agents_ (described below) and the _built-in node_. The built-in node is a node that exists within the controller process.

Reviewer: ```suggestion
Jenkins supports two types of nodes: _agents_ (described below) and the _built-in node_.
The built-in node is a node that exists within the controller process.
```
one sentence per line for easier review

> The built-in node is a node inside the controller process, hence the name. 
> It can have executors, but it is not an agent. Agents and the built-in node are nodes. 
> All agents are nodes, but not all nodes are agents. Executors exist, are available or busy, 
> and can be added or removed, but they are not "run".

Reviewer 2: ```suggestion
Jenkins can use two kinds of nodes: _agents_ (descibed below) and the _built-in node_. The built-in node is a node inside the controller process.
Both agents and built-in node can be used to run tasks, though running tasks on built-in node is link:/book/security/controller-isolation/[discouraged because of security, performance and scalability reasons].
Node's ability to run tasks is determined by the number of executors configured for the node.
To disable running tasks on built-in node you can set the number of executors to 0.
```
I think the bit about performance/security implications of using built-in node is still valid and should not be removed. The sentence about executors was just an explanation of why "running an executor" was an incorrect phrase in the old version -- explaining relation between executors and nodes here may make sense though (

Author: Thank you for the suggestions.

> Both agents and the built-in node can be used to run tasks, though running tasks on the built-in node is discouraged for security, performance, and scalability reasons.

Reviewer: ```suggestion
Both agents and the built-in node can be used to run tasks.
However, running tasks on the built-in node is discouraged for security, performance, and scalability reasons.
```
separate thoughts to make it a bit more readable

A node is taken offline if any of these values
go outside the configured threshold.

The Jenkins controller itself runs on a special _built-in node_. 

> The built-in node is a node inside the controller process. It is possible to run agents with it and it can have executors. 

Reviewer 3: The first two sentences are directly contradicting each other.

The third sentence is wrong. The built-in node explicitly is _not_ an agent.

> although Executors exist, are available or busy, and can be added or removed 

Reviewer 3: Sentence starts with lowercase letter.

Also, this content from the linked issue was meant mostly to describe reality, not be copy & pastable into the docs.

> but this can degrade performance, reduce scalability of the Jenkins instance, and create serious security problems

Reviewer 3: Unclear how this relates to the start of the sentence.

Author: >Also, this content from the linked issue was meant mostly to describe reality, not be copy & pastable into the docs.
>
I dont know much about core jenkins internals so what should i use as reference ?
if there is a document can you give me the link,this would also increase my knowledge of jenkins internals

Reviewer 3: > I dont know much about core jenkins internals so what should i use as reference ?
if there is a document can you give me the link,this would also increase my knowledge of jenkins internals

You're writing the documentation here. If we had it, it wouldn't need to be written 😉
